File: 351001/orsik/publisher/src/main/java/com/sergey/orsik/service/AuthService.java
package com.sergey.orsik.service;

import com.sergey.orsik.dto.request.LoginRequestTo;
import com.sergey.orsik.dto.response.AuthTokenResponseTo;
import com.sergey.orsik.entity.Creator;
import com.sergey.orsik.repository.CreatorRepository;
import com.sergey.orsik.security.JwtService;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.core.AuthenticationException;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

import static org.springframework.http.HttpStatus.UNAUTHORIZED;

@Service
public class AuthService {

    private final AuthenticationManager authenticationManager;
    private final CreatorRepository creatorRepository;
    private final JwtService jwtService;

    public AuthService(
            AuthenticationManager authenticationManager,
            CreatorRepository creatorRepository,
            JwtService jwtService) {
        this.authenticationManager = authenticationManager;
        this.creatorRepository = creatorRepository;
        this.jwtService = jwtService;
    }

    public AuthTokenResponseTo authenticate(LoginRequestTo request) {
        try {
            authenticationManager.authenticate(
                    new UsernamePasswordAuthenticationToken(request.getLogin(), request.getPassword())
            );
        } catch (AuthenticationException ex) {
            throw new ResponseStatusException(UNAUTHORIZED, "Invalid credentials");
        }
        Creator creator = creatorRepository.findByLogin(request.getLogin())
                .orElseThrow(() -> new ResponseStatusException(UNAUTHORIZED, "Invalid credentials"));
        String token = jwtService.generateToken(creator.getLogin(), creator.getRole());
        return new AuthTokenResponseTo(token, "Bearer");
    }
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/service/AuthService.kt
package com.example.distcomp.service

import com.example.distcomp.dto.request.LoginRequestTo
import com.example.distcomp.dto.response.AuthTokenResponseTo
import com.example.distcomp.dto.response.CurrentUserResponseTo
import com.example.distcomp.repository.CreatorRepository
import com.example.distcomp.security.CurrentUserService
import com.example.distcomp.security.JwtService
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.stereotype.Service

@Service
class AuthService(
    private val authenticationManager: AuthenticationManager,
    private val creatorRepository: CreatorRepository,
    private val jwtService: JwtService,
    private val currentUserService: CurrentUserService
) {
    fun login(request: LoginRequestTo): AuthTokenResponseTo {
        authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(request.login, request.password)
        )

        val creator = creatorRepository.findByLogin(request.login)
            ?: throw BadCredentialsException("Invalid login or password")

        return AuthTokenResponseTo(accessToken = jwtService.generateToken(creator))
    }

    fun currentUser(): CurrentUserResponseTo {
        val creator = currentUserService.currentCreator()
        return CurrentUserResponseTo(
            id = creator.id ?: throw BadCredentialsException("Authenticated creator id not found"),
            login = creator.login.orEmpty(),
            role = creator.role.name,
            firstName = creator.firstname,
            lastName = creator.lastname
        )
    }
}
